File: math/armstrong_number/Python/armstrongnumber.py
import math
number = int(input("Enter any number: "))
count = 0
digit_sum = 0
for i in str(number):
    count += 1
for i in str(number):
    digit_sum += pow(int(i), count)

if digit_sum == number:
    print("The number is armstrong")
else:
    print("How about NOPE")

File: math/armstrong_number/Swift/ArmstrongNumber.swift
import Foundation

func checkArmstrongNumber(_ num: Int) -> Bool {
	var sum = 0
	var tempNum = num
	var reminder = 0
	let digits = String(num).count
	while tempNum != 0  {
		reminder = tempNum % 10
		sum = sum + Int(pow(Double(reminder), Double(digits)))
		tempNum /= 10
		if sum > num {
			break
		}
	}
	if sum == num {
		return true
	}
	return false
}

print("Pick a number to check: ", terminator: "")
let input = readLine()
print()

if let input = input {
	if let num = Int(input) {
		let result = checkArmstrongNumber(num)
		let text = result ? "is" : "is not"
		print("\(input) \(text) an Armstrong Number")
	} else {
		print("Please enter a number")
	}
} else {
	print("Please enter a value")
}
